src/components/icon/icon.spec.js

```diff
@@ -419,6 +419,29 @@ describe('mdIcon service', function() {
         $scope.$digest();
       });
 
+      describe('and the URL is a data URL', function() {
+        var svgData = '<svg><g><circle r="50" cx="100" cy="100"></circle></g></svg>';
+
+        describe('and the data is base64 encoded', function() {
+          it('should return correct SVG markup', function() {
+            var data = 'data:image/svg+xml;base64,' + btoa(svgData);
```

Comment:
I added a function expression test in my last round of updates (in response to your initial set of comments). You can see it here: https://github.com/angular/material/pull/7547/files#diff-c132fcb7e577f14138c70b13aff4ea4fR253